### PR TITLE
Implement Already Dead Banishing from previous chain link

### PR DESCRIPTION
**Pull request by PvtVoid**
Big commits so I put it on it's own branch.

### CODE REVIEW

File: CardDictionaries/HitEffects.php [hunk 1]
```diff
@@ -91,9 +91,16 @@ function EVOHitEffect($cardID)
...
+        if(IsHeroAttackTarget()) {
+          $deck = new Deck($defPlayer);
+          if($deck->Empty()) { WriteLog("The opponent deck is already... depleted."); break; }
+            $deck->BanishTop(banishedBy:$mainPlayer);
+            AddDecisionQueue("SEARCHCOMBATCHAIN", $mainPlayer, "-");
+            AddDecisionQueue("SETDQCONTEXT", $mainPlayer, "Choose which card to banish");
+            AddDecisionQueue("CHOOSECARDID", $mainPlayer, "<-");
+            AddDecisionQueue("ALREADYDEAD", $mainPlayer, "CC,-," . $mainPlayer, 1);
+          }
+          break;      case "EVO241":
```
Comment (OotTheMonk) [inline CardDictionaries/HitEffects.php:103]: Minor thing, but it looks like a newline is missing here

File: CardLogic.php [hunk 2]
```diff
@@ -1334,25 +1334,12 @@ function ProcessTrigger($player, $parameter, $uniqueID, $target="-", $additional
         $index = FindCharacterIndex($player, "HVY648");
         CharacterChooseSubcard($player, $index, isMandatory:false);
         AddDecisionQueue("ADDDISCARD", $player, "-", 1);
-        AddDecisionQueue("MULTIZONEINDICES", $player, "THEIRITEMS:minCost=0;maxCost=1", 1);
+        AddDecisionQueue("MULTotherPlayerIZONEINDICES", $player, "THEIRITEMS:minCost=0;maxCost=1", 1);
```
Comment (OotTheMonk) [inline CardLogic.php:1337]: I feel like this otherPlayer here is probably a typo that would freeze the game

File: Constants.php [hunk 3]
```diff
@@ -193,6 +193,7 @@ function ChainLinksPieces()
 //5 - Hit on link
 //6 - Modified Base Attack Stats - e.g. Transmogrify
 //7 - Modal Play Ability - e.g. Enlightened Strike
+//8 - Defending Cards
```
Comment (OotTheMonk) [inline Constants.php:196]: You shouldn't need to add defending cards to chainLinkSummary, there's already a $chainLinks that stores the cards still on the combat chain (see the above function)

**Review by OotTheMonk — CHANGES_REQUESTED** [review]
Added a few comments

File: CardLogic.php (hunk 2 above)
Comment (PvtVoid) [reply CardLogic.php:1337]: Definitely a typo! Fixed

File: CardDictionaries/HitEffects.php (hunk 1 above)
Comment (PvtVoid) [reply CardDictionaries/HitEffects.php:103]: I added the missing break line

File: Constants.php (hunk 3 above)
Comment (PvtVoid) [reply Constants.php:196]: I never realized the frame for this kind of interactions was already implemented. I thought it needed to be started from the ground up, considering it was never implemented to work 100%. 

I've removed the info from chainLinkSummary and use ChainLinks properly

**Review by OotTheMonk — APPROVED** [review]

**Pull request merged**